### PR TITLE
Improvements for Remove-AadAppsForBc 

### LINKED ISSUE #3472: Remove-AadAppsForBc doesn't remove AAD App for Other Services
**Describe the issue**
New-AadAppsForBc' creates an AAD app for other services, but won't delete it if you use `Remove-AadAppsForBc'.

**Scripts used to create container and cause the issue**
```PowerShell
$bcAuthContext = New-BcAuthContext [...]

New-AadAppsForBc `
  -appIdUri https://dockerhub.company.com/test `
  -bcAuthContext $bcAuthContext `
  -IncludeApiAccess `
  -IncludeExcelAadApp `
  -IncludeOtherServicesAadApp `
  -preAuthorizePowerShell

Remove-AadAppsForBc `
  -appIdUri https://dockerhub.company.com/test `
  -bcAuthContext $bcAuthContext
```

**Full output of scripts**
```
BcContainerHelper version 6.0.14
BC.HelperFunctions emits usage statistics telemetry to Microsoft
Authenticated as app [...]
Creating AAD App for WebClient
Creating AAD App for API Access
Creating AAD App for Other Services

Remove AAD App for Api
Remove AAD App for Excel
Remove AAD App for PowerBI
Remove AAD App for EMail Service
...
```

**Screenshots**

**Additional context**
- does it happen all the time?
- did it use to work?

### PR DESCRIPTION
**Pull request by pri-kise**
Use SecureString for AccessToken in Remove-AadAppsForBc if necessary:
I added a check to use Convert the AccessToken to a SecureString if it's required for `Connect-MgGraph` .

and 
Fix #3472

### CODE REVIEW

**Review by freddydk — APPROVED** [review]

**Comment by freddydk** [comment]
Thanks

File: AzureAD/Remove-AadAppsForBc.ps1 [hunk 1]
```diff
@@ -39,6 +39,16 @@ try {
...
     }
 
+    # Check the AccessToken since Microsoft Graph V2 requires a SecureString
+    $graphAccesTokenParameter = (Get-Command Connect-MgGraph).Parameters['AccessToken']
+  
+    if ($graphAccesTokenParameter.ParameterType -eq [securestring]){
+        $useSecureStringForAccessToken = $true
+        if ($accessToken) {
+            $accessToken = ConvertTo-SecureString -String $accessToken -AsPlainText -Force
+        }
```
Comment (freddydk) [inline AzureAD/Remove-AadAppsForBc.ps1:49]: You convert the accesstoken to a securestring here and then again later
I guess lines 47-49 should be removed?

**Review by freddydk — CHANGES_REQUESTED** [review]

File: AzureAD/Remove-AadAppsForBc.ps1 (hunk 1 above)
Comment (pri-kise) [reply AzureAD/Remove-AadAppsForBc.ps1]: You're right. This happens when you refactor the code several times during development.

File: AzureAD/Remove-AadAppsForBc.ps1 [hunk 2]
```diff
@@ -47,16 +54,19 @@ try {
...
+            if ($useSecureStringForAccessToken){
+                Connect-MgGraph -AccessToken (ConvertTo-SecureString -String $accessToken -AsPlainText -Force) | Out-Null
             }
             else {
-                Connect-MgGraph
+                Connect-MgGraph -AccessToken $accessToken | Out-Null
             }
         }
+        else {
+            Connect-MgGraph -Scopes 'Application.ReadWrite.All' | Out-Null
```
Comment (freddydk) [inline AzureAD/Remove-AadAppsForBc.ps1:68]: This comes into play when you have not specified useCurrentMicrosoftGraphConnection and not specified a bcAuthContext nor an accesstoken.
What does it do?

Comment (pri-kise) [reply AzureAD/Remove-AadAppsForBc.ps1:68]: This tries to connect with Microsoft Graph via UI Interaction/Login via Browser and requests this scope via delegated user login. 
I only copied this from the the other cmdlet `New-AadAppsForBc`

**Review by freddydk — APPROVED** [review]

**Pull request merged**